debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/Replicator.java

```diff
@@ -380,7 +380,7 @@ protected boolean performInitialSync() {
             return false;
         }
 
-        if (collections.isEmpty()) {
+        if (collections.isEmpty() && isInitialSyncExpected()) {
```

Comment:
Shouldn't the new flag be used here?

Reply:
@gunnarmorling Yes it should.  I also made a few changes locally as well to flush out that last commit a bit more inline with what I believe we were talking about.  I'll rebase & open a PR that supersedes this one shortly.

Reply:
@gunnarmorling Actually I redact my earlier statement.  I don't think that call should be there at all for two reasons:

1. We actually only write the log entry for MongDB if we do an initial sync.
2. Calling that method at this point after the sync will always return `false`.

I think its safe for the mongodb implementation to just check if collections is empty after the snapshot has finished for now and we leave the remaining as a common framework port concern, wdyt @gunnarmorling ?